### PR TITLE
replace selenium by cypress

### PR DESCRIPTION
**Pull request by zekth**
closes: https://github.com/qmk/qmk_configurator/issues/471
Easier to deploy, to develop, to use.
anyway it's better.

Et voilà

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,6 +7,8 @@
     "build": "vue-cli-service build",
     "lint": "vue-cli-service lint",
     "test:e2e": "vue-cli-service test:e2e",
```
Comment (yanfali) [inline package.json]: remove this?

File: package.json [hunk 2]
```diff
@@ -7,6 +7,8 @@
...
+    "test:cypress": "cypress run",
```
Comment (yanfali) [inline package.json]: what does this task do? It doesn't run standalone

File: tests/integration/start.spec.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      '30wer'
+    ]);
+  });
+  it('Starts the app Check the dom', function() {
+    cy.visit('/');
+    cy.get('#controller-top').should('be.visible');
+    cy.get('#visual-keymap').should('be.visible');
+    cy.get('#status').should('be.visible');
+    cy.get('#keycodes-section').should('be.visible');
+    cy.get('.openSettings').click();
```
Comment (yanfali) [inline tests/integration/start.spec.js]: split this into it's own test.

File: tests/integration/start.spec.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    cy.get('.settings-panel', { timeout: 5000 }).should('be.visible');
+    cy.get('.slideout-panel-bg').click();
+    cy.get('#testkeys').click();
```
Comment (yanfali) [inline tests/integration/start.spec.js]: same, should be it's own test. You're changing the route

File: package.json (hunk 2 above)
Comment (zekth) [reply package.json]: It could but easier to have everything starting directly from yarn

Comment (yanfali) [reply package.json]: ok, I think I see what you mean now. Maybe put a comment on this.

File: src/electron.js [hunk 5]
```diff
@@ -2,12 +2,14 @@ import store from './store';
 
 export default {
   init() {
-    store.commit('app/enableElectron')
+    store.commit('app/enableElectron');
```
Comment (yanfali) [inline src/electron.js]: should we move this inside the guard?

File: tests/integration/start.spec.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+      '1upkeyboards/1up60hte',
+      '1upkeyboards/1up60rgb',
+      '1upkeyboards/super16',
+      '1upkeyboards/sweet16',
+      '2_milk',
+      '30wer'
+    ]);
+    cy.visit('/');
+  });
+  it('Top Controller should be instanciated and visible', function() {
```
Comment (yanfali) [inline tests/integration/start.spec.js]:
```suggestion
  it('Top Controller should be instantiated and visible', function() {
```

**Comment by yanfali** [comment]
@e11i0t23 heads up, we're moving the electron code behind a guard to check for the presence of window.Bridge. Let us know if this causes you any issues. Thanks.

**Pull request merged**